gettext-maps/build.rs

```diff
@@ -0,0 +1,149 @@
+    use fish_mo_file_parser::parse_mo_file;
+    use std::{
+        fs::File,
+        io::{BufWriter, Read, Write},
+    };
+
+    let po_dir = PathBuf::from(MANIFEST_DIR).parent().unwrap().join("po");
+
+    // Ensure that the directory is created, because clippy cannot compile the code if the
+    // directory does not exist.
+    std::fs::create_dir_all(mo_file_dir).unwrap();
+
+    let localization_map_path =
+        Path::new(&env::var("OUT_DIR").unwrap()).join("localization_maps.rs");
+    let mut localization_map_file = BufWriter::new(File::create(&localization_map_path).unwrap());
+
+    // This will become a map which maps from language identifiers to maps containing localizations
+    // for the respective language.
+    let mut catalogs = phf_codegen::Map::new();
+
+    let all_langs_enabled = std::env::var("CARGO_FEATURE_ALL_LANGUAGES").is_ok();
+
+    for dir_entry_result in po_dir.read_dir().unwrap() {
+        let dir_entry = dir_entry_result.unwrap();
+        let po_file_path = dir_entry.path();
+        let lang = po_file_path
+            .file_stem()
+            .expect("All entries in the po directory must be regular files.");
+        let language = lang.to_str().unwrap().to_owned();
+
+        if !all_langs_enabled
+            && std::env::var(format!("CARGO_FEATURE_{}", language.to_uppercase())).is_err()
+        {
+            // Localizations for this language are not enabled.
+            // The `CARGO_FEATURE_*` variables are set by cargo based on which features are enabled.
+            // The feature names are converted to uppercase and `-` is replaced by `_`.
+            continue;
+        }
+        let mo_file_path = mo_file_dir.join(lang);
+
+        let update_mo_file = || {
+            // Try to create a new MO file.
+            std::process::Command::new("msgfmt")
+                .arg("--check-format")
+                .arg(format!("--output-file={}", mo_file_path.to_str().unwrap()))
+                .arg(&po_file_path)
+                .output()
+        };
+
+        if let Ok(metadata) = std::fs::metadata(&mo_file_path) {
+            // MO file exists, but might be outdated.
+            let mo_mtime = metadata.modified().unwrap();
+            let po_mtime = dir_entry.metadata().unwrap().modified().unwrap();
+            if mo_mtime < po_mtime {
+                update_mo_file().unwrap();
+            };
+        } else {
+            // MO file does not exist.
+            update_mo_file().unwrap();
+        }
+
+        // At this point we have an up-to-date MO file.
+        // We proceed to parse the file into a map, which is put into a Rust source file,
+        // such that it can be used from other source code.
+
+        let mut mo_file = File::open(&mo_file_path).unwrap();
+        let mut file_content = vec![];
+        mo_file.read_to_end(&mut file_content).unwrap();
+        let language_localizations = parse_mo_file(&file_content).unwrap();
```

Comment:
do we actually want intermediate `*.mo` files?
Maybe the `*.po` files contain the same information;
I guess `*.mo` file parsing might be more robust?
But since we generate the *.po files ourselves, we have a tight grip on the exact format, so maybe it doesn't matter?